test/cmocka/src/math/trig/lut_sin_16b_fixed.c

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2024 Intel Corporation. All rights reserved.
+//
+// Author: Slawomir Blauciak <slawomir.blauciak@linux.intel.com>
+// Author: Seppo Ingalsuo <seppo.ingalsuo@linux.intel.com>
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <math.h>
+#include <cmocka.h>
+
+#include <sof/audio/format.h>
+#include <sof/math/lut_trig.h>
+
+#include "trig_tables.h"
+
+#define CMP_TOLERANCE	3.1e-5
+#define _M_PI		3.14159265358979323846	/* pi */
+
+static void test_math_trig_lut_sin_fixed(void **state)
+{
+	(void)state;
+
+	int theta;
+
+	for (theta = 0; theta < 360; ++theta) {
+		double rad = _M_PI * (theta / 180.0);
```

Comment:
hopefully the compiler will calculate `_M_PI / 180.0` at compile time, but parentheses might actually prevent it from doing that and make it a (redundant) run-time calculation

Reply:
It's cmocka test code so we don't care about performance even if there would be emulated floats.